sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -65,12 +65,15 @@ object StringUtils extends Logging {
     "(?s)" + out.result() // (?s) enables dotall mode, causing "." to match new lines
   }
 
-  private[this] val trueStrings = Set("t", "true", "y", "yes", "1").map(UTF8String.fromString)
-  private[this] val falseStrings = Set("f", "false", "n", "no", "0").map(UTF8String.fromString)
+  private[this] val trueStrings =
+    Set("t", "true", "y", "yes", "1", "on").map(UTF8String.fromString)
+
+  private[this] val falseStrings =
+    Set("f", "false", "n", "no", "0", "off").map(UTF8String.fromString)
```

Comment:
It seems only PostgreSQL accepts `on` and `off`?

Reply:
Yes I guess so. Do you know other common string representattion used in other databases?

Reply:
But PostgreSQL also accepts`of`, `tru`, `fals`, ...:
```sql
postgres=# select cast('of' as boolean), cast('tru' as boolean), cast('fals' as boolean);
 bool | bool | bool
------+------+------
 f    | t    | f
(1 row)
```

https://github.com/postgres/postgres/commit/9729c9360886bee7feddc6a1124b0742de4b9f3d

Reply:
Ah okay. Let me add that too. Thank you